src/main/java/com/mojang/brigadier/suggestion/SuggestionsBuilder.java

```diff
@@ -8,18 +8,27 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 import java.util.concurrent.CompletableFuture;
 
 public class SuggestionsBuilder {
     private final String input;
+    private final String inputLowerCase;
     private final int start;
     private final String remaining;
+    private final String remainingLowerCase;
     private final List<Suggestion> result = new ArrayList<>();
 
-    public SuggestionsBuilder(final String input, final int start) {
+    public SuggestionsBuilder(final String input, final String inputLowerCase, final int start) {
         this.input = input;
+        this.inputLowerCase = inputLowerCase;
         this.start = start;
         this.remaining = input.substring(start);
+        this.remainingLowerCase = inputLowerCase.substring(start);
```

Comment:
It might not be the case that for calling toLowerCase on a string preserves the number of chars in the string, so using the start index there could cause the cached value to be offset. Better to use `this.remaining.toLowerCase(Locale.ROOT)`

Reply:
This is true https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#toUpperCase-java.util.Locale- but I wonder if there are other places in brigadier that depends on the assumption that a lowercased string has the same length as the original string and maps 1 to 1. If there is, only fixing this location doesn't solve the problem, and I recommend doing such a fix in a separate pull request instead.

Reply:
For Brigadier it looks like only `SuggestionsBuilder` (and code using it) performs case conversion. I have created #92 for the issue you described.